src/allvars.h

```diff
@@ -1,3421 +1,3425 @@
+    MyFloat AGS_Dt_Psi_Re;
+    MyFloat AGS_Gradients_Psi_Re[3];
+    MyFloat AGS_Gradients2_Psi_Re[3][3];
+    MyFloat AGS_Psi_Im;
+    MyFloat AGS_Psi_Im_Pred;
+    MyFloat AGS_Dt_Psi_Im;
+    MyFloat AGS_Gradients_Psi_Im[3];
+    MyFloat AGS_Gradients2_Psi_Im[3][3];
+    MyFloat AGS_Dt_Psi_Mass;
+#endif
+#endif
+#if defined(AGS_FACE_CALCULATION_IS_ACTIVE)
+    MyLongDouble NV_T[3][3];                                           /*!< holds the tensor used for gradient estimation */
+#endif
+}
+ *P,				/*!< holds particle data on local processor */
+ *DomainPartBuf;		/*!< buffer for particle data used in domain decomposition */
+
+
+#ifndef GDE_LEAN
+#define GDE_TIMEBEGIN(i) (P[i].a0)
+#define GDE_VMATRIX(i, a, b) (P[i].V_matrix[a][b])
+#define GDE_INITDENSITY(i) (P[i].init_density)
+#else
+#define GDE_TIMEBEGIN(i) (All.TimeBegin)
+#define GDE_VMATRIX(i, a, b) (0.0)
+#define GDE_INITDENSITY(i) (All.GDEInitStreamDensity)
+#endif
+
+#if defined(BLACK_HOLES)
+#define BPP(i) P[(i)]
+#endif
+
+
+#if defined(HYDRO_TENSOR_FACE_CORRECTIONS_NGBITER)
+#define HYDRO_TENSOR_FACE_CORRECTIONS_NUMBER_MOMWTS 15
+#else
+#define HYDRO_TENSOR_FACE_CORRECTIONS_NUMBER_MOMWTS 9
+#endif
+
+
+/* the following struture holds data that is stored for each SPH particle in addition to the collisionless
+ * variables.
+ */
+extern struct sph_particle_data
+{
+    /* the PRIMITIVE and CONSERVED hydro variables used in STATE reconstruction */
+    MyDouble Density;               /*!< current baryonic mass density of particle */
+#ifdef HYDRO_MESHLESS_FINITE_VOLUME
+    MyDouble MassTrue;              /*!< true particle mass ('mass' now is -predicted- mass */
+    MyDouble dMass;                 /*!< change in particle masses from hydro step (conserved variable) */
+    MyDouble DtMass;                /*!< rate-of-change of particle masses (for drifting) */
+    MyDouble GravWorkTerm[3];       /*!< correction term needed for hydro mass flux in gravity */
+    MyDouble ParticleVel[3];        /*!< actual velocity of the mesh-generating points */
+#endif
+
+    MyDouble Pressure;              /*!< current pressure */
+    MyDouble InternalEnergy;        /*!< specific internal energy [internal thermal energy per unit mass] of cell */
+    MyDouble InternalEnergyPred;    /*!< predicted value of the specific internal energy at the current time */
+    //MyDouble dInternalEnergy;     /*!< change in specific internal energy from hydro step */ //manifest-indiv-timestep-debug//
+    MyDouble DtInternalEnergy;      /*!< rate of change of specific internal energy */
+
+    MyDouble VelPred[3];            /*!< predicted SPH particle velocity at the current time */
+    //MyDouble dMomentum[3];        /*!< change in momentum from hydro step (conserved variable) */ //manifest-indiv-timestep-debug//
+    MyDouble HydroAccel[3];         /*!< acceleration due to hydrodynamical force (for drifting) */
+    MyDouble GradVelNorm;           /* Magnitude of Velocity Gradient*/

```

Comment:
If you use a local variable when writing the output, then this is not necessary. Removing this from particle_data reduces memory usage slightly, but it won't make much of a practical difference.